File: 388/388_version2.py
class Solution:
	def lengthLongestPath(self,input:str) -> int:
            files = input.split("\n")
            path = {0:0}
            max_len = 0
            for f in files:
                level = f.count("\t")
                file_len = len(f) - level
                if "." in f:
                    max_len = max([max_len,path[level]+file_len])
                else:
                    path[level+1] = path[level]+file_len+1
            return max_len

File: 388/388_version2.java
class Solution{
	public int lengthLongestPath(String input){
        	HashMap<Integer,Integer> path = new HashMap<>();
        	path.put(0,0);// initialize the hashmap
        	int maxLen = 0;
        	for(String s:input.split("\n")){
            		int level = s.lastIndexOf("\t")+1;
            		int fileLen = s.length()-level;
            		if(s.contains("."))
                		maxLen = Math.max(path.get(level)+fileLen,maxLen);
            		else
                		path.put(level+1,path.get(level)+fileLen+1);
        	}
        	return maxLen;
    	}
}
